docs/ECN-test-plan.md

```diff
@@ -0,0 +1,145 @@
+- On SONiC DUT configure the following:
+  1. A single lossless priority Pi (0 <= i <= 7).
+  2. Configure minimum and maximum ECN marking threshold of the lossless priority Pi to N KB (e.g., 100 KB).
+  3. DUT should have enough buffer to hold 2N KB packets.
+
+- Configure following traffic items on the Keysight device:
+  1. Test data traffic: A traffic item from the Keysight Tx port to
+        the Keysight Rx port. This traffic item should have DSCP value mapped to lossless priority Pi configured in the DUT. Number of packets should be fixed to 2N (e.g., 200), size of each packet should be 1KB each.
+  2. PFC PAUSE storm: Persistent PFC pause frames from the Keysight
+        Rx port to the Keysight Tx port. The priorities of PFC pause
+        frames should be same as that of 'Test data traffic'. And the
+        inter-frame transmission interval should be lesser than
+        per-frame pause duration.
+
+#### Test Steps
+
+1. Start PFC PAUSE storm to fully block the lossless priority at the
+    DUT.
+2. From the Keysight Tx port, send 2N data packets to the receiver. The packets should be mapped to priority Pi on the DUT.
+3. After all the test data packets are sent, start data capture on the Keysight Rx port. Then, stop PFC PAUSE.
+4. Stop capture after all packets are received.
+5. Verify the following:
+   * Keysight Rx port must receive 2N test data packets.
+   * The first test data packet received by Keysight Rx port should be ECN marked.
+   * The last test data packet received by Keysight Rx port should not be ECN marked.
+5. Repeat the test with all the lossless priorities.
+
+
+### Test Case #2 - ECN marking accuracy
+
+#### Test Objective
+
+This test aims to verify the DUT’s ECN marking accuracy, e.g., if the actual ECN marking probability is close to the theoretical value.
+
+#### Test Configuration
+
+- On SONiC DUT configure the following:
+  1. A single lossless priority Pi (0 <= i <= 7).
+  2. Configure the minimum ECN marking threshold, the maximum ECN marking threshold, and the maximum marking probability of the Priority Pi to Kmin KB, Kmax KB, and Pmax % respectively. Let the Kmin, Kmax, and Pmax be 500 KB, 2000 KB, and 5% respectively for the first iteration.
+  3. DUT should have enough buffer to hold (Kmax+10) packets each of 1KB size.
+
+- Configure following traffic items on the Keysight device:
+  1. Test data traffic: A traffic item from the Keysight Tx port to
+        the Keysight Rx port. This traffic item should have DSCP value mapped to lossless priority Pi configured in the DUT. It should also have fixed number of packets, Kmax+10 (e.g., 2010), each having 1KB frame size.
+  2. PFC PAUSE storm: Persistent PFC pause frames from the Keysight
+        Rx port to the Keysight Tx port. The priorities of PFC pause
+        frames should be the same as that of 'Test data traffic'. And the
+        inter-frame transmission interval should be lesser than the 
+        per-frame pause duration.
+
+#### Test Steps
+
```

Comment:
We should start packet capturing before sending data packets. See https://github.com/baiwei0427/sonic-mgmt/blob/ixia/tests/ixia/ECN_ACCURACY_README.md

Reply:
Updated.